### DISCUSSION TITLE
Add dnf distro-sync as system upgrade options

### DISCUSSION THREAD

**Discussion by AngryPenguinPL** [timlau/yumex-ng#96] [Ideas]
Hi.
We plan to make yumex-ng the default package management application in our distribution for GNOME.

We only have one problem. Well, we do not use the "dnf update" command to update the system because it sometimes causes problems for us, because our roots come from urpmi and rpm5, hence we sometimes have epoch or api version in libname in our packages. We try to systematically eliminate it in subsequent releases, but it causes problems that when using "dnf update" sometimes a package conflict may occur or a package may be omitted because dnf will recognize it as an older version than in the repo (because it was epoch). The "dnf distro-sync" command solves these problems and is the command we recommend to our users.

Hence the question: is it possible to add an update function using "dnf distro-sync"?

**Comment by timlau** [comment]
what yumex does is showing the available updates and let you select the packages you what to update.

distro-sync is kind of special case, where you both update and downgrade packages so the installed ones matches the ones available in repos, even if the installed ones has a higher version than the ones in the repositories.
 
The normal use case is after upgrading to a new distro version with other repos, there contain some packages that has an old version than the ones from the previous distro version, not something a normal user need on daily base. So it is a little hard for me to see the use-case for the normal user on day to day use.

From a technical point of view the dnf4/dnf5daemon  has no public API for doing a distro-sync AFAIK.

dnf5daemon has public API for doing a distro-sync of a list of packages, so it is would be technincal possible to do a distro sync

don't know what version af dnf you are using in your distro (what distro is it ??)

currently yumex support both dnf5 and dnf4, when dnf5 become default in fedora and the versions that uses dnf4 as default, become EOL, I will properly deprecate support for dnf4.

As long as both dnf4 & 5 is supported, I will properly not add any functionality there cant work with both backends, because it makes it hard to maintain and as yumex is spare-time project, I have to chose where to spend my limited time to where it has the most benefit.

making some kind of disto-sync all button could be an option, like the update all button in the flatpak view.

**Comment by timlau** [comment]
@AngryPenguinPL  better late, than never , system distro.sync has been added now
